packages/serverless/src/domain/use-case/collection.ts

```diff
@@ -49,6 +53,37 @@ async function createCollection(
   }
 }
 
+async function readCollectionInfo(
+  databaseName: string,
+  collectionName: string
+): Promise<Collection> {
+  const indexes = await ModelCollection.getInstance(
+    databaseName,
+    collectionName
+  ).listIndexes();
+  const schema = await getSchemaDefinition(databaseName, collectionName);
+  const ownership = await readMetadata(
+    databaseName,
+    collectionName,
+    null,
+    ZKDATABASE_USER_NOBODY
+  );
+
+  return { name: collectionName, indexes, schema, ownership };
+}
+
+async function listCollections(databaseName: string): Promise<Collection[]> {
+  const collectionNames =
+    await ModelDatabase.getInstance(databaseName).listCollections();
+
+  const collections: Collection[] = await Promise.all(
```

Comment:
Use `@orochi-network/queue` and use Fill instead. Better error handling

Reply:
done